Benchmark/main.cpp

```diff
@@ -0,0 +1,174 @@
+#include <QGuiApplication>
+#include <algorithm>
+#include <chrono>
+#include <iostream>
+#include <vector>
+
+void myMessageOutput(QtMsgType type,
+                     [[maybe_unused]] const QMessageLogContext& context,
+                     const QString& msg)
+{
+    QByteArray localMsg = msg.toLocal8Bit();
+    QString typeStr = "OTHER";
+    switch (type)
+    {
+        case QtDebugMsg:
+            typeStr = "DBG";
+            break;
+        case QtInfoMsg:
+            typeStr = "INF";
+            break;
+        case QtWarningMsg:
+            typeStr = "WRN";
+            break;
+        case QtCriticalMsg:
+            typeStr = "CRT";
+            break;
+        case QtFatalMsg:
+            typeStr = "FTL";
+            break;
+    }
+    std::cout << QString("%1: %2")
+                     .arg(typeStr)
+                     .arg(localMsg.constData())
+                     .toStdString()
+              << std::endl;
+}
+
+void interpolatorMeasurement(int led_number, int led_size, int angle,
+                             const std::string& pixmap_path, QObject* q_obj)
+{
+    QPixmap pix_map{};
+    pix_map.load(QString::fromStdString(pixmap_path));
+    Interpolator interpolator_obj{q_obj};
+
+    interpolator_obj.setPixmap(&pix_map);
+
+    QPoint rot_centr(pix_map.width() / 2, pix_map.height() / 2);
+
+    QRect rect{QPoint{static_cast<int>(led_size * 0.5),
+                      static_cast<int>(led_size * (-0.5))},
+               QSize{led_size, led_size}};
+
+    for (int rot = 0; rot < 360; rot += angle)
+    {
+        for (int i = 0; i < led_number; i++)
+        {
+            rect.moveTo(rect.topLeft() + QPoint{led_size, 0});
+
+            QColor color = interpolator_obj.interpolatorSetLedColor(
+                interpolator_obj.interpolatorTransform(
+                    Transform{rot_centr, angle}, rect));
+        }
+    }
+}
+
+using algorithm = void (*)(int, int, int, const std::string&, QObject*);
+
+int measureStatisticTime(int iteration, algorithm fun, int a, int b, int c,
+                         const std::string& pixmap_path, QObject* q_obj)
+{
+    std::vector<double> average_cont{};
+
+    for (int i = 0; i < iteration; i++)
+    {
+        auto begin = std::chrono::high_resolution_clock::now();
+        fun(a, b, c, pixmap_path, q_obj);
+        auto end = std::chrono::high_resolution_clock::now();
+        auto elapsed =
+            std::chrono::duration_cast<std::chrono::microseconds>(end - begin);
+        average_cont.push_back(elapsed.count());
+    }
+
+    std::sort(average_cont.begin(), average_cont.end());
+    average_cont.erase(average_cont.begin());
+    average_cont.erase(average_cont.end() - 1);
+
+    double sum{};
+    for (double dd : average_cont)
+    {
+        sum += dd;
+    }
+    sum /= average_cont.size();
+    return sum;
+}
+
+void argParsing(int argc, char* argv[], int& led_number, int& led_size,
+                int& angle, int& iteration, std::string& pixmap_path)
```

Comment:
I would suggest to return struct with ready parameters than putting everything as out-ref params.

If some parameters are not set then you could assign default values. For main algorithm it will be transparent if parameters are given from command line or are loaded from defaults.

Reply:
Corrected.